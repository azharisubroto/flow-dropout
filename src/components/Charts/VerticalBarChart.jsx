import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import ReactTooltip from 'react-tooltip'
import styles from './VerticalBarChart.module.scss'

const VerticalBarChart = ({ label, barValue, item, type, adjList }) => {
  const [renderTooltip, setRenderTooltip] = useState(false)

  useEffect(() => {
    setRenderTooltip(true)
  }, [])

  return (
    <div className={styles.VerticalBarChart}>
      <div
        className={`${styles['hor-bar-chart']} ${styles[[type]]}`}
        style={{
          height: `${barValue ? barValue + '%' : 'auto'}`
        }}
        data-for={item.id}
        data-tip={`Type: ${type}<br/>Branch: ${item.id}<br/>adjList: ${adjList}`}>
        <div className={styles['hbc-label']}>{label}</div>
      </div>

      {/* Tool Tip */}
      {renderTooltip && <ReactTooltip id={item.id} place="top" type="dark" effect="float" multiline={true} />}

      <div className={styles['hbc-value']}>{barValue}%</div>
    </div>
  )
}

VerticalBarChart.propTypes = {
  adjList: PropTypes.array,
  background: PropTypes.string,
  barValue: PropTypes.number,
  item: PropTypes.any,
  label: PropTypes.string,
  type: PropTypes.string
}

export default VerticalBarChart
