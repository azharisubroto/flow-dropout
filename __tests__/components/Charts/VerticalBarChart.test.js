import React from 'react'
import { shallow } from 'enzyme'
import VerticalBarChart from '@/components/Charts/VerticalBarChart'

describe('<VerticalBarChart />', () => {
  let wrapper, VerticalBarChartComp

  beforeEach(() => {
    VerticalBarChartComp = (
      <VerticalBarChart
        label={`Label`}
        item={'Item Name'}
        barValue={70}
        type={'Type Name'}
        adjList={['node1', 'node2']}
      />
    )

    wrapper = shallow(VerticalBarChartComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct classes', () => {
    expect(wrapper.find('.hor-bar-char-wrapper')).toHaveLength(1)
    expect(wrapper.find('.hor-bar-chart')).toHaveLength(1)
    expect(wrapper.find('.hbc-label')).toHaveLength(1)
    expect(wrapper.find('.hbc-value')).toHaveLength(1)
  })

  it('should render correct percentage value', () => {
    expect(wrapper.find('.hbc-value').text()).toEqual('70%')
  })

  it('should render correct label', () => {
    expect(wrapper.find('.hbc-label').text()).toEqual('Label')
  })
})
