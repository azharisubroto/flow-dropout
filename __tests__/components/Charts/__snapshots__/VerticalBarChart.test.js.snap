// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalBarChart /> should renders correctly 1`] = `ShallowWrapper {}`;
